sci-libs/cantera/cantera-2.4.0.ebuild

```diff
@@ -0,0 +1,130 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{4,5,6} )
+
+inherit desktop python-single-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+cti cxx fortran pch +python static-libs test"
+
+REQUIRED_USE="
+	|| ( cxx python )
+	cti? ( ${PYTHON_REQUIRED_USE} )
+	fortran? ( cxx )
+	python? ( cti )
+	static-libs? ( cxx )
+	^^ ( $(python_gen_useflags 'python3*') )
```

Comment:
You've got only py3 now, so this is redundant.

Reply:
I tried to remove 'python3*' parameter earlier but then I receive an error:

```
# repoman manifest
 * ERROR: sci-libs/cantera-2.4.0::local failed (depend phase):
 *   _python_impl_matches: takes at least 2 parameters
 * 
 * Call stack:
 *                 ebuild.sh, line 635:  Called source '/usr/local/portage/sci-libs/cantera/cantera-2.4.0.ebuild'
 *   cantera-2.4.0.ebuild, line  26:  Called python_gen_useflags
 *   python-single-r1.eclass, line 350:  Called _python_impl_matches 'python3_4'
 *    python-utils-r1.eclass, line 179:  Called die
 * The specific snippet of code:
 *      [[ ${#} -ge 2 ]] || die "${FUNCNAME}: takes at least 2 parameters"
```
The same error for `repoman -dx full`

Reply:
Replace the whole thing with `${PYTHON_REQUIRED_USE}` as generated by eclass.

Reply:
it's strange but if I set `^^ ( ${PYTHON_REQUIRED_USE} )` then I get an error:
```
# emerge -1av =cantera-2.4.0

These are the packages that would be merged, in order:

Calculating dependencies \

!!! Problem resolving dependencies for =sci-libs/cantera-2.4.0
... done!

!!! The ebuild selected to satisfy "=cantera-2.4.0" has unmet requirements.
- sci-libs/cantera-2.4.0::local USE="cti cxx fortran pch python static-libs -test" ABI_X86="(64)" PYTHON_SINGLE_TARGET="python3_6 -python3_4 -python3_5" PYTHON_TARGETS="python3_6 -python3_4 -python3_5"

  The following REQUIRED_USE flag constraints are unsatisfied:
    exactly-one-of ( exactly-one-of ( python_single_target_python3_4 python_single_target_python3_5 python_single_target_python3_6 ) python_single_target_python3_6? ( python_targets_python3_6 ) )

  The above constraints are a subset of the following complete expression:
    any-of ( cxx python ) cti? ( exactly-one-of ( python_single_target_python3_4 python_single_target_python3_5 python_single_target_python3_6 ) python_single_target_python3_4? ( python_targets_python3_4 ) python_single_target_python3_5? ( python_targets_python3_5 ) python_single_target_python3_6? ( python_targets_python3_6 ) ) fortran? ( cxx ) python? ( cti ) static-libs? ( cxx ) exactly-one-of ( exactly-one-of ( python_single_target_python3_4 python_single_target_python3_5 python_single_target_python3_6 ) python_single_target_python3_4? ( python_targets_python3_4 ) python_single_target_python3_5? ( python_targets_python3_5 ) python_single_target_python3_6? ( python_targets_python3_6 ) )
```
while I have active only `python_targets_python3_6` and `python_single_target_python3_6`.
Or did I misunderstood the required replacement?

Reply:
It already contains `^^`.

Reply:
Thanks. It works now. I noticed that as before changing this line I could set additional `python_targets_python3_x`. Is it normal? `${EPYTHON}` will be set by `python_single_target_python3_6` if I have it as active?

Reply:
Done